### PR TITLE
Merge results from all plugins

### PR DESCRIPTION
**Pull request by lukaszo**
With multiple plugins selected Genie was returning only result of the last plugin. Now merged result from all plugins will be returned.

### CODE REVIEW

**Comment by lukaszo** [comment]
I'm not 100% sure about this approach. Maybe you have another idea how to implement this?

**Comment by satyaranjanp** [comment]
We will update you after reviewing the changes.

File: genie/genie-controller.go [hunk 1]
```diff
@@ -603,3 +615,77 @@ func getK8sPodAnnotations(client *kubernetes.Clientset, k8sArgs utils.K8sArgs) (
...
+			break
+		}
+	}
+
+	for _, route := range result.Routes {
+		if route.GW == nil {
+			if defaultGw == nil {
+				return nil, fmt.Errorf("Route has not set GW and the default GW is missing: %v", result)
+			}
+			route.GW = defaultGw
```
Comment (satyaranjanp) [inline genie/genie-controller.go]: This statement will assign the value into 'route' variable only; it will not get assigned into 'result'.

Comment (lukaszo) [reply genie/genie-controller.go]: Routes is a list of pointers to Route structure https://github.com/containernetworking/cni/blob/master/pkg/types/current/types.go#L140 it's visible in result

**Comment by sushanthakumar** [comment]
LGTM

**Comment by sushanthakumar** [comment]
@lukaszo, changes are fine, can u pls rebase once to resolve conflict so that we can merge this PR, thanks

**Comment by lukaszo** [comment]
@sushanthakumar that's super cool. In the meantime the patch become bigger but I didn't push it here because I didn't know if you are interested in merging in. I will prepare new version soon.

It's bigger because some CNI plugins return result in wrong way and it requires some fixing before merge with previous result.

**Comment by sushanthakumar** [comment]
@lukaszo , u can update using latest changes and we can have a review

**Comment by lukaszo** [comment]
@sushanthakumar I've update PR. I can also add some unit tests if it's required.

**Pull request merged**